File: stepik-union/src/main/java/org/stepik/plugin/actions/step/AbstractStepAction.java
package org.stepik.plugin.actions.step;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.stepik.core.StepikProjectManager;
import org.stepik.core.actions.StudyActionWithShortcut;
import org.stepik.core.courseFormat.StepNode;
import org.stepik.core.courseFormat.StudyNode;

import javax.swing.*;

/**
 * @author meanmail
 */
abstract class AbstractStepAction extends StudyActionWithShortcut {
    AbstractStepAction(@Nullable String text, @Nullable String description, @Nullable Icon icon) {
        super(text, description, icon);
    }

    @Nullable
    @Contract("null -> null")
    static StepNode getCurrentStep(@Nullable Project project) {
        StudyNode<?, ?> studyNode = StepikProjectManager.getSelected(project);
        return studyNode instanceof StepNode ? (StepNode) studyNode : null;
    }

    @Override
    public void update(@NotNull AnActionEvent e) {
        StepNode stepNode = getCurrentStep(e.getProject());
        e.getPresentation().setEnabled(stepNode != null && !stepNode.getWasDeleted());
    }
}

File: stepik-union/src/main/kotlin/org/stepik/core/actions/step/AbstractStepAction.kt
package org.stepik.core.actions.step

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.project.Project
import org.jetbrains.annotations.Contract
import org.stepik.core.ProjectManager
import org.stepik.core.actions.StudyActionWithShortcut
import org.stepik.core.courseFormat.StepNode
import javax.swing.Icon


abstract class AbstractStepAction protected constructor(text: String?, description: String?, icon: Icon?) :
        StudyActionWithShortcut(text, description, icon) {

    override fun update(e: AnActionEvent?) {
        val presentation = e?.presentation ?: return
        val stepNode = getCurrentStep(e.project)
        presentation.isEnabled = stepNode != null && !stepNode.wasDeleted
    }

    companion object {
        @Contract("null -> null")
        fun getCurrentStep(project: Project?): StepNode? {
            if (project == null) {
                return null
            }
            val projectManager = ServiceManager.getService(project, ProjectManager::class.java)
            return projectManager.getSelected(project)
        }
    }
}
